File: Array/0121.Best-Time-to-Buy-and-Sell-Stock/Solution.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        
        int maxProfit = 0, minPrice = int.MaxValue;

        foreach(int price in prices) {
            maxProfit = Math.Max(maxProfit, price-minPrice);
            minPrice = Math.Min(price, minPrice);
        }
        return maxProfit;
    }
}

File: Array/0121.Best-Time-to-Buy-and-Sell-Stock/Solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = numeric_limits<int>::max(), maxGain = 0;
        for (int price : prices) {
            minPrice = min(minPrice, price);
            maxGain = max(maxGain, price-minPrice);
        }
        return maxGain;
    }
};
